checker.cpp

```diff
@@ -1,46 +1,73 @@
 #include <iostream>
-#include <assert.h>
 #include <unistd.h>
+#include <cassert>
+
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
-    }
+struct Alert {
+    string message;
+    int delayInSeconds;
+};
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+void showAlertWithDelay(const Alert &alert) {
+    cout << alert.message << endl;
+    for (int i = 0; i < 6; i++) {
+        cout << "\r* " << flush;
+        sleep(alert.delayInSeconds);
+        cout << "\r *" << flush;
+        sleep(alert.delayInSeconds);
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+}
+
+bool isOutOfRange(float value, float lowerLimit, float upperLimit) {
+    return value < lowerLimit || value > upperLimit;
+}
+
+bool isTemperatureCritical(float temperature) {
+    return isOutOfRange(temperature, 95, 102);
+}
+
+bool isPulseRateOutOfRange(float pulseRate) {
+    return isOutOfRange(pulseRate, 60, 100);
+}
+
+bool isSpo2OutOfRange(float spo2) {
+    return isOutOfRange(spo2, 0, 90);
+}
+
+void checkAndAlert(const Alert &alert, bool condition) {
+    if (condition) {
+        showAlertWithDelay(alert);
     }
-    return 0;
-  }
-  return 1;
 }
 
-int main() {
-  assert(!vitalsOk(99, 102, 70));
-  assert(vitalsOk(98.1, 70, 98));
-  cout << "Done\n";
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+    Alert temperatureAlert = {"Temperature critical!", 1};
+    Alert pulseRateAlert = {"Pulse Rate is out of range!", 1};
+    Alert spo2Alert = {"Oxygen Saturation out of range!", 1};
+
+    checkAndAlert(temperatureAlert, isTemperatureCritical(temperature));
+    checkAndAlert(pulseRateAlert, isPulseRateOutOfRange(pulseRate));
+    checkAndAlert(spo2Alert, isSpo2OutOfRange(spo2));
+
+    return !temperatureAlert.message.empty() ||
```

Comment:
The message is initialized above, irrespective of in-range or out-of-range. That's why the return is not changing and the assert is failing?